bigtable/write_samples.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Import google bigtable client lib
+require "google/cloud/bigtable"
+
+def write_simple project_id, instance_id, table_id
+  bigtable = Google::Cloud::Bigtable.new project_id: project_id
+
+  #  [START bigtable_writes_simple]
+  table = bigtable.table instance_id, table_id
+  column_family = "stats_summary"
+  timestamp = Time.now.to_i * 1000
```

Comment:
This value needs to be changed from milliseconds to microseconds with "millisecond granularity". 

The correct expression is: `(Time.now.to_f * 1000).to_i * 1000`

For example: `1564257960168000`


Cell `timestamp_micros` values are indeed in microseconds but must match the granularity of the table, which is currently always milliseconds, meaning that the microsecond values must be rounded to the nearest 1000.

See googleapis/google-cloud-ruby#3709 for the documentation fix in the client library.

Reply:
Another option would be to use `Numeric#round` passing `-3`:

```ruby
(Time.now.to_f * 1000).round #=> 1564432211441
(Time.now.to_f * 1000).round(-3) #=> 1564432211000
```

Reply:
Yes, that's a probably easier to read.

Reply:
It may also be more accurate/precise to encourage use Rational (`Time#to_r`) over Float (`Time#to_f`):

```ruby
(Time.now.to_r * 1000000).round(-3) #=> 1564257960168123
```

Reply:
FWIW, my very simple benchmarks shows using `Time#to_f` is 2x faster than `Time#to_r`.

Reply:
I changed the expression to `(Time.now.to_f * 1000000).round(-3)` in googleapis/google-cloud-ruby#3709.